### DISCUSSION TITLE
How to crop only images with specific dimension (width_x_height)

### DISCUSSION THREAD

**Discussion by john-soda** [ImageMagick/ImageMagick#2455] [Help]
Is it possible to crop or resize only images with the same dimension. All images are in the same folder. Is it possible to do this only with ImageMagick or do I need to write a script?

**Comment by snibgo** [comment]
What version of IM, on what platform?

What are "images with the same dimension"? I don't understand what you want to do.

In general, a single IM command can read an image, clone it, crop or resize that clone, then delete either the original or the cropped clone according to some condition, then write the image.

**Comment by john-soda** [comment]
With specific Dimension I mean.
1920 x 588 -> crop
1920 x 588 -> crop
1920 x 620
1919 x 588
1920 x 588 -> crop

My platform is linux. Manjaro with Plasma. IM version 7.0.10

**Comment by GeeMack** [comment]
With ImageMagick v7 you can use FX expressions within the command to crop based on the dimensions. This command will read the input image, and only if the dimensions are 1920x588 it will crop it to 400x300. If the input is not 1920x588, the dimensions of the output image will not be changed.

    wide=400
    high=300
    magick input.png \
       -crop "%[fx:w==1920&&h==588?${wide}:w]x%[fx:w==1920&&h==588?${high}:h]+0+0" result.png

If the X and Y offsets for the crop are not "+0+0", you can use FX expressions there, too.

Keep in mind if you're using a lossy format like JPG, the command will process and write the output file even if it doesn't modify the dimensions.

Learn more about [FX Expressions at this LINK](https://imagemagick.org/script/fx.php).
